tests/integration/test_sharding.py

```diff
@@ -583,3 +623,329 @@ def test_asking_follower(cluster):
     conn.send_command('get', 'key')
     with raises(ResponseError, match="ASK"):
         conn.read_response()
+
+
+def test_cluster_slots_for_single_slot_sg(cluster):
+    cluster.create(3, raft_args={'sharding': 'yes', 'external-sharding': 'yes'})
+    cluster_stable_shardgroup_id = "1" * 32
+    cluster_importing_shardgroup_id = "2" * 32
+    cluster_migrating_shardgroup_id = "3" * 32
+
+    assert cluster.execute(
+        'RAFT.SHARDGROUP', 'REPLACE',
+        '3',
+
+        cluster_stable_shardgroup_id,
+        '1', '1',
+        '0', '0', '1', '123',
+        '%s00000001' % cluster_stable_shardgroup_id, '1.1.1.1:1111',
+
+        cluster_importing_shardgroup_id,
+        '1', '1',
+        '501', '501', '2', '123',
+        '%s00000001' % cluster_importing_shardgroup_id, '2.2.2.2:2222',
+
+        cluster_migrating_shardgroup_id,
+        '1', '1',
+        '501', '501', '3', '123',
+        '%s00000001' % cluster_migrating_shardgroup_id, '3.3.3.3:3333',
+        ) == b'OK'
+
+    def validate_slots(cluster_slots):
+        assert len(cluster_slots) == 2
+
+        stable_node_id = "{}00000001".format(cluster_stable_shardgroup_id).encode()
+        migrate_node_id = "{}00000001".format(cluster_migrating_shardgroup_id).encode()
+
+        for i in range(len(cluster_slots)):
+            node_id = cluster_slots[i][2][2]
+            start_slot = cluster_slots[i][0]
+            end_slot = cluster_slots[i][1]
+
+            if node_id == stable_node_id:
+                assert start_slot == 0, cluster_slots
+                assert end_slot == 0, cluster_slots
+            elif node_id == migrate_node_id:
+                assert start_slot == 501, cluster_slots
+                assert end_slot == 501, cluster_slots
+            else:
+                assert False, "failed to match id {}".format(node_id)
+
+    validate_slots(cluster.node(1).client.execute_command('CLUSTER', 'SLOTS'))
```

Comment:
Btw, if you feel like using in future, just fyi, we have shorter alternatives to execute a command:

`cluster.execute('CLUSTER', 'SLOTS')`    (sends it to the leader node, retries command on failure)
or
`cluster.node(1).execute('CLUSTER', 'SLOTS')` .

Reply:
except if you want to catch MOVED as an expected situation cluster.execute() handles MOVED itself